packages/assertion-monitor/index.ts

```diff
@@ -0,0 +1,193 @@
+import { PublicClient, createPublicClient, http } from 'viem'
+import yargs from 'yargs'
+import { ChildNetwork as ChainInfo } from '../utils'
+import { getChainFromId } from './chains'
+import { reportAssertionMonitorErrorToSlack } from './reportAssertionMonitorAlertToSlack'
+
+const options = yargs(process.argv.slice(2))
+  .options({
+    configPath: { type: 'string', default: 'config.json' },
+    enableAlerting: { type: 'boolean', default: false },
+  })
+  .strict()
+  .parseSync()
+
+const configFileContent = fs.readFileSync(
+  path.join(process.cwd(), options.configPath),
+  'utf-8'
+)
+const config = JSON.parse(configFileContent)
+
+if (!Array.isArray(config.childChains) || config.childChains.length === 0) {
+  console.error('Error: Chains not found in the config file.')
+  process.exit(1)
+}
+const nodeCreatedEventAbi = {
+  type: 'event',
+  name: 'NodeCreated',
+  inputs: [
+    {
+      type: 'uint64',
+      name: 'nodeNum',
+      indexed: true,
+    },
+    {
+      type: 'bytes32',
+      name: 'parentNodeHash',
+      indexed: true,
+    },
+    {
+      type: 'bytes32',
+      name: 'nodeHash',
+      indexed: true,
+    },
+    {
+      type: 'bytes32',
+      name: 'executionHash',
+      indexed: false,
+    },
+    {
+      type: 'tuple',
+      name: 'assertion',
+      components: [
+        {
+          type: 'tuple',
+          name: 'beforeState',
+          components: [
+            {
+              type: 'tuple',
+              name: 'globalState',
+              components: [
+                {
+                  type: 'bytes32[2]',
+                  name: 'bytes32Vals',
+                },
+                {
+                  type: 'uint64[2]',
+                  name: 'u64Vals',
+                },
+              ],
+            },
+            {
+              type: 'uint8',
+              name: 'machineStatus',
+            },
+          ],
+        },
+        {
+          type: 'tuple',
+          name: 'afterState',
+          components: [
+            {
+              type: 'tuple',
+              name: 'globalState',
+              components: [
+                {
+                  type: 'bytes32[2]',
+                  name: 'bytes32Vals',
+                },
+                {
+                  type: 'uint64[2]',
+                  name: 'u64Vals',
+                },
+              ],
+            },
+            {
+              type: 'uint8',
+              name: 'machineStatus',
+            },
+          ],
+        },
+        {
+          type: 'uint64',
+          name: 'numBlocks',
+        },
+      ],
+      indexed: false,
+    },
+    {
+      type: 'bytes32',
+      name: 'afterInboxBatchAcc',
+      indexed: false,
+    },
+    {
+      type: 'bytes32',
+      name: 'wasmModuleRoot',
+      indexed: false,
+    },
+    {
+      type: 'uint256',
+      name: 'inboxMaxCount',
+      indexed: false,
+    },
+  ],
+} as const
+
+const getLogsForLast7Days = async (
```

Comment:
In general, the lesser we hardcode in logic, the better it would be for dev-experience of the teams who configure and use these scripts for their use-cases.